### PR TITLE
pugpugpug

### PR DESCRIPTION
**Pull request by elessar-ch**
image
image
image
image
image

### CODE REVIEW

**Comment by elessar-ch** [comment]
Ok, seriously. I upgraded from the primordial version 0.0.3!! of Jade to the current version 2.0.0 beta6 of Pug.
I also needed to upgrade some other libraries in the metalsmith ecosystem around our docs build and had to do some refactoring especially regarding the filters.
Looks like it's working pretty well now and I don't expect a lot of averse side effects.
Also the build-time of the Jade/Pug pages is now around half as long as before.

File: docs/page/404.pug [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 title: Oh no! Page not found!
-template: page.jade
+layout:: page.pug
```
Comment (AndyOGo) [inline docs/page/404.pug]: Why the double `::` colons?

File: docs/page/changelog.pug [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
 title: What's new
-template: page.jade
+layout:: page.pug
```
Comment (AndyOGo) [inline docs/page/changelog.pug]: Same double `::` here

**Comment by elessar-ch** [comment]
Thx @AndyOGo. Copypasta 🍝 error. Will fix right away

**Comment by davidknezic** [comment]
OMG this is great! And such a hilarious PR! 😂 

I'm still gonna die rebasing this to the webpack branch 🔫 . Nonetheless, let's merge!

**Comment by elessar-ch** [comment]
@davidknezic Rebase?

@sventschui @davidknezic: does one of you guys know why we compiled our jade-pages and then had another step to fit them with a layout? Did jade at that point not support a layout functionality or did/does it not suit our requirements?

They have the extends functionality which let's you basically define some containers in a layout and overwrite them in your file: https://pugjs.org/language/extends.html
I can imagine using this for the pug files would speed up the build a little bit as well. For markdown files we'd still need the other step though.

**Comment by davidknezic** [comment]
In the webpack branch, I moved the docs to another folder. There will be lots of conflicts, when this is merged into develop. 😢 But hey, that's life.

About the layout, do you really think it would be faster? Because now the layout is built once and applied to every document. Otherwise the layout would be parsed for every file, or am I mistaken?

**Comment by elessar-ch** [comment]
Oh yeah right :P
Not sure how pug handles the layouts. I'd imagine they're doing some caching as well. I just noticed that the step to apply the layout isn't the fastest either. I'll give it a shot and see what's happening.

**Comment by elessar-ch** [comment]
Uhh... that's waaaayyy slower... I'll leave it at that.

**Comment by davidknezic** [comment]
Okay, do you want to merge then?

**Comment by AndyOGo** [comment]
awesome 👍

**Pull request merged**